### PR TITLE
Complete GET All Articles endpoint

### PR DESCRIPTION
**Pull request by karolinakonior**

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -78,4 +78,15 @@ describe('GET /api', () => {
...
     })
+})
+
+describe('GET /api/articles', () => {
+    test('GET: 200 sends and array of articles to the client sorted by date in descending order', () => {
+        return request(app)
+        .get('/api/articles')
+        .expect(200)
+        .then((response) => {
+            expect(response.body.articles).toBeSortedBy('created_at', { descending: true });
```
Comment (Jodiech) [inline __tests__/app.test.js:89]: Good use of jest-sorted in your test here 👍

File: __tests__/app.test.js [hunk 2]
```diff
@@ -78,4 +78,15 @@ describe('GET /api', () => {
...
+        })
```
Comment (Jodiech) [inline __tests__/app.test.js:90]: Great start to testing, you need to add a few more please:
- A test to check that this responds with an array of article objects.
- What errors do you need to test for here?

File: controllers/app.controllers.js [hunk 3]
```diff
@@ -17,4 +18,13 @@ exports.getEndpoints = (req, res, next) => {
     .catch((err) => {
         next(err)
     })
+}
+
+exports.getArticles = (req, res, next) => {
+    fetchArticles().then((articles) => {
+        res.status(200).send({ articles })
```
Comment (Jodiech) [inline controllers/app.controllers.js:25]: Nice to see you are returning back on the appropriate key of articles 👍

**Review by Jodiech — CHANGES_REQUESTED** [review]
Great work on this ticket so far, however, there are some cases which haven't been tested for yet (please see the comments) Please submit another PR when you're done! Let me know If you need any info. 🚀

File: __tests__/app.test.js [hunk 4]
```diff
@@ -87,6 +87,20 @@ describe('GET /api/articles', () => {
         .expect(200)
         .then((response) => {
             expect(response.body.articles).toBeSortedBy('created_at', { descending: true });
+            expect(typeof response.body.articles).toBe('object');
```
Comment (Augs0) [inline __tests__/app.test.js:90]: This is a good attempt and you're very close. We are dealing with many articles and will need to test all of them rather tha just the first one. How could we use iteration here to help us?

File: controllers/app.controllers.js [hunk 5]
```diff
@@ -22,7 +22,11 @@ exports.getEndpoints = (req, res, next) => {
 
 exports.getArticles = (req, res, next) => {
     fetchArticles().then((articles) => {
-        res.status(200).send({ articles })
+        if (articles.length === 0) {
```
Comment (Augs0) [inline controllers/app.controllers.js:25]: This _could_ happen but is not really an error to concern ourselves with. If you want to keep this, it should go in the model and be a Promise.reject that leads to the error handler in your app

**Review by Augs0 — CHANGES_REQUESTED** [review]
Almost there, Karolina! I'd like you to edit your test slightly so we are testing all articles please

**Pull request merged**